modules/standard/IO.chpl

```diff
@@ -2401,12 +2396,12 @@ proc openfd(fd: fd_t, hints:iohints=IOHINT_NONE, style:iostyle = defaultIOStyle(
   var err:syserr = ENOERR;
   var ret = openfd(fd, err, hints, style);
   if err {
-    var path:c_string_copy;
+    var path_cs:c_string_copy;
     var e2:syserr = ENOERR;
-    e2 = qio_file_path_for_fd(fd, path);
-    if e2 then path = c"unknown";
-    // FIX ME: could use a toString() that doesn't allocate space
-    ioerror(err, "in openfd", path:string);
+    e2 = qio_file_path_for_fd(fd, path_cs);
+    var path = if e2 then "unknown"
+                     else new string(path_cs, needToCopy=false);
```

Comment:
I find this version - and several other cases like it - a little bit surprising. I'd expect the path_cs memory to be deallocated at the end of this block. But aren't you putting that pointer in to the new string and not copying it?
I think that setting path_cs = nil (or whatever the equivalent that works is) would solve the problem.
Note that some of these error handling cases are probably not properly tested right now.